pkg/secrets/passphrase/manager.go

```diff
@@ -151,7 +153,35 @@ func NewPassphaseSecretsManagerFromState(state json.RawMessage) (secrets.Manager
 	// This is not ideal, but we don't have a great way to prompt the user in this case, since this may be
 	// called during an update when trying to read stack outputs as part servicing a StackReference request
 	// (since we need to decrypt the deployment)
-	phrase := os.Getenv("PULUMI_CONFIG_PASSPHRASE")
+	phrase := ""
+	passphrase, isOk := os.LookupEnv("PULUMI_CONFIG_PASSPHRASE")
+	if isOk {
+		phrase = passphrase
+	}
+	phraseFile, isOk := os.LookupEnv("PULUMI_CONFIG_PASSPHRASE_FILE")
+	if isOk {
+		phraseFilePath, err := filepath.Abs(phraseFile)
+		if err != nil {
+			return nil, errors.Wrap(err, "unable to detect passphrase path")
+		}
+
+		phraseDetails, err := ioutil.ReadFile(phraseFilePath)
+		if err != nil {
+			return nil, errors.Wrap(err, "unable to read PULUMI_CONFIG_PASSPHRASE_FILE")
+		}
+
+		phrase = strings.TrimSpace(string(phraseDetails))
+	}
+
+	// At this point, we don't know if it's an incorrect passphrase. We only know if there is a passphrase or there is
+	// not. Ideally, we would prompt the user for the passphrase at this point but we can't do that in the CLI is in an
+	// update operation, so we should at least error out with an appropriate message here to ensure that the user
+	// understands why the operation fails unexpectedly.
+	if phrase == "" {
+		return nil, errors.New("unable to find either `PULUMI_CONFIG_PASSPHRASE` nor " +
+			"`PULUMI_CONFIG_PASSPHRASE_FILE` when trying to access the Passphrase Secrets Manager; please ensure one " +
```

Comment:
The feature is called "secrets provider" right?  https://www.pulumi.com/docs/intro/concepts/secrets/#initializing-a-stack-with-alternative-encryption